cmd/ipfs/kubo/daemon.go

```diff
@@ -572,6 +572,9 @@ take effect.
 		}
 	}()
 
+	// Clear any cached offline node and set the online daemon node
+	// This ensures HTTP RPC server uses the online node, not any cached offline node
+	cctx.ClearCachedNode()
```

Comment:
:information_source: TLDR: there was no bug in autoconfig, it was FX bug.

I've spent ages checking every integration point of AutoConfig, but it turned out the failure was due to FX race condition that was always present, but never manifested. 

IIUC this was always broken, and things worked only by sheer luck (correct node was picked every time), but when we shifted execution order due to autoconfig cache check, the incorrect node (temporary offline one)  was picked up by RPC, effectively bricking the `daemon`. 

This explicit cleanup should ensure that `ipfs daemon`-triggered initializations beyond this stage are using correct (new) node.